uint8 player_id
Predicate[] goal
---
uint8 executor_id
